#984 grafan,prometheus setup (#1048)
layer5io/meshery: docs/pages/guides/meshery-metrics.md

---
layout: guide
title: meshery metrics
description: How to use meshery metrics

> permalink: guides/mesherymetrics

Reviewer: ```suggestion
permalink: guides/meshery-metrics
```

type: guide
---

> Inorder to generate reports/load testing of your k8 applications, meshery uses `Grafana` and `Prometheus`.

Reviewer: ```suggestion
In order to generate report/load testing of your k8s applications, Meshery uses `Grafana` and `Prometheus`.
```

> Here you can find steps to generate these endpoints in different environemnts/service mesh adaptors.

Reviewer: ```suggestion
Here you can find steps to generate these endpoints in different environments or for different service mesh adapters.
```

* TOC
{:toc}

### Istio + Minikube(dev env)

In order for Meshery to gain access to grafana and prometheus, their network ports need to be exposed.

> **1) Get minikube ip**

Reviewer: Can we use markdown syntax here ?

Author: if we use markdown syntax it will be added to table contents that's why I used bold.
let me know if there any possibility to skip subheadings in TOC

Get Minikube ip by executed below command.
    
```        
    $minikube ip
    o/p: 172.17.0.2
```
<i>Note: Here Istio is installed in `istio-system` name space and `productpage` sample app is installed in `book-info` name space.
</i>

**2)Expose Prometheus service**

> a)By default  `prometheus`  spec type is configured to `ClusterIP`  you can chage it to `NodePort` using below command.

Reviewer: ```suggestion
a)By default  `prometheus`  spec type is configured to `ClusterIP`  you can change it to `NodePort` using below command.
```

Reviewer: Also, can we use an unordered list here rather than a, b, c sub listings?

Author: yep, we can try that in the next version.

```
$kubectl patch svc prometheus -p '{"spec": {"type": "NodePort"}}' -n istio-system
```

b)Get NodePort of `prometheus` service using below command

```
$kubectl describe services prometheus -n istio-system|grep NodePort

o/p:NodePort:  http  30535/TCP
```

c) Prometheus endpoint will be http://$MINIKUBE_IP:NODE_PORT

```
http://172.17.0.2:30822
```

**3)Expose Grafana service**

> a)By default  `Grafana`  spec type is configured to `ClusterIP`  you can chage it to `NodePort` using below command.

Reviewer: ```suggestion
a)By default  `Grafana`  spec type is configured to `ClusterIP`  you can change it to `NodePort` using below command.
```

```
$kubectl patch svc grafana -p '{"spec": {"type": "NodePort"}}' -n istio-system
```

b)Get NodePort of `Grafana` service using below command

```
$kubectl describe services grafana -n istio-system|grep NodePort

o/p:NodePort:  http  32130/TCP
```

c) Grafana endpoint will be http://$MINIKUBE_IP:NODE_PORT

```
http://172.17.0.2:32130
```

> **4)Expose  istio sample app service productpage**

Reviewer: ```suggestion
**4)Expose Istio sample app service product-page**
```

> a)By default  `productpage`  spec type is configured to `ClusterIP`  you can chage it to `NodePort` using below command.

Reviewer: ```suggestion
a)By default  `productpage`  spec type is configured to `ClusterIP`  you can change it to `NodePort` using below command.
```

```
$kubectl patch svc grafana -p '{"spec": {"type": "NodePort"}}' -n book-info
```

b)Get NodePort of `productpage` service using below command

```
$kubectl describe services productpage -n book-info|grep NodePort

o/p:NodePort:  http  30535/TCP
```

> c) productpage endpoint will be http://$MINIKUBE_IP:NODE_PORT

Reviewer: ```suggestion
c) productpage endpoint will be available at http://$MINIKUBE_IP:NODE_PORT
```
